widget/embedded/src/components/SwapDetails/SwapDetails.helpers.tsx

```diff
@@ -0,0 +1,215 @@
+                    (step.status === 'waitingForApproval' &&
+                      !isNetworkStatusInWarningState(step))
+                      ? 'loading'
+                      : index === urls.length - 1 &&
+                        (index !== 0 ||
+                          (index === 0 &&
+                            explorerUrl.description !== 'Approve')) &&
+                        step.status === 'failed'
+                      ? 'error'
+                      : 'success'
+                  }
+                  title={
+                    !explorerUrl.description
+                      ? i18n.t('View transaction')
+                      : `${explorerUrl.description
+                          .substring(0, 1)
+                          .toUpperCase()}${explorerUrl.description.substring(
+                          1
+                        )} Tx`
+                  }
+                  action={
+                    explorerUrl.url && (
+                      <Button
+                        variant="ghost"
+                        dense
+                        onClick={() => window.open(explorerUrl.url, '_blank')}>
+                        <LinkIcon size={16} />
+                      </Button>
+                    )
+                  }
+                />
+              </Fragment>
+            );
+          })}
+          {step.status !== 'failed' && isNetworkStatusInWarningState(step) && (
+            <Alert
+              type="warning"
+              title={message.shortMessage}
+              action={
+                (!!switchNetwork ||
+                  showNetworkModal ===
+                    PendingSwapNetworkStatus.WaitingForConnectingWallet) && (
+                  <Button
+                    size="micro"
+                    type="warning"
+                    onClick={() => {
+                      if (!!switchNetwork) {
+                        setNetworkModal(
+                          PendingSwapNetworkStatus.WaitingForNetworkChange
+                        );
+                        switchNetwork?.().catch((e) => {
+                          console.log(e);
+                        });
+                      } else {
+                        setNetworkModal(
+                          PendingSwapNetworkStatus.WaitingForConnectingWallet
+                        );
+                      }
+                    }}>
+                    {showNetworkModal ===
+                    PendingSwapNetworkStatus.WaitingForConnectingWallet
+                      ? i18n.t('Connect')
+                      : i18n.t('Change')}
+                  </Button>
+                )
+              }
+            />
+          )}
+          {step.status === 'failed' && (
+            <Alert
+              type="error"
+              title={message.shortMessage}
+              action={
+                message.detailedMessage.content && (
+                  <Tooltip content={message.detailedMessage.content}>
```

Comment:
We need a better approach for displaying error details.

Reply:
fixed.

Reply:
fixed.